--- ./decompiled/net/minecraft/bootstrap/Bootstrap.java
+++ ./src/main/java/net/minecraft/bootstrap/Bootstrap.java
@@ -242,13 +242,15 @@
             });
     }
 
     public void startLauncher(File launcherJar) {
         println("Starting launcher.");
         try {
-            Class aClass = new URLClassLoader(new URL[] { launcherJar.toURI().toURL() }).loadClass("net.minecraft.launcher.Launcher");
+            URLClassLoader launcherClassLoader = new URLClassLoader(new URL[] { launcherJar.toURI().toURL() });
+            com.simon816.minecraft.launcher.Injector.start(this, workDir, launcherClassLoader, proxy);
+            Class aClass = launcherClassLoader.loadClass("net.minecraft.launcher.Launcher");
             Constructor constructor = aClass.getConstructor(new Class[] { JFrame.class, File.class, Proxy.class, PasswordAuthentication.class, java.lang.String[].class, Integer.class });
             constructor.newInstance(new Object[] { this, this.workDir, this.proxy, this.proxyAuth, this.remainderArgs, Integer.valueOf(5) });
         } catch (Exception e) {
             throw new FatalBootstrapError(new StringBuilder().append("Unable to start: ").append(e).toString());
         }
     }
